pkg/xray/watches.go

```diff
@@ -0,0 +1,267 @@
+	if len(resources.Resources) == 0 {
+		return []interface{}{}
+	}
+
+	var list []interface{}
+	for _, res := range resources.Resources {
+		resourceMap := map[string]interface{}{}
+		resourceMap["type"] = res.Type
+		if len(res.Name) > 0 {
+			resourceMap["name"] = res.Name
+		}
+		if len(res.BinaryManagerId) > 0 {
+			resourceMap["bin_mgr_id"] = res.BinaryManagerId
+		}
+		resourceMap["filter"] = packFilters(res.Filters)
+		list = append(list, resourceMap)
+	}
+
+	return list
+}
+
+func packFilters(filters []WatchFilter) []interface{} {
+	if filters == nil {
+		return []interface{}{}
+	}
+
+	var l []interface{}
+	for _, f := range filters {
+		m := map[string]interface{}{
+			"type":  f.Type,
+			"value": f.Value,
+		}
+		l = append(l, m)
+	}
+
+	return l
+}
+
+func packAssignedPolicies(policies []WatchAssignedPolicy) []interface{} {
+	if policies == nil {
+		return []interface{}{}
+	}
+
+	var l []interface{}
+	for _, p := range policies {
+		m := make(map[string]interface{})
+		m["name"] = p.Name
+		m["type"] = p.Type
+		l = append(l, m)
+	}
+
+	return l
+}
+
+func resourceXrayWatchCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := unpackWatch(d)
+	_, err := m.(*resty.Client).R().SetBody(watch).Post("xray/api/v2/watches")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(watch.GeneralData.Name)
+	resourceXrayWatchRead(ctx, d, m)
+	return diags
+}
+
+func resourceXrayWatchRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := Watch{}
+	resp, err := m.(*resty.Client).R().SetResult(&watch).Get("xray/api/v2/watches/" + d.Id())
+	if err != nil {
+		if resp != nil && resp.StatusCode() == http.StatusNotFound {
+			log.Printf("[WARN] Xray watch (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diags
+	}
+	if err := d.Set("description", watch.GeneralData.Description); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("active", watch.GeneralData.Active); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("watch_resource", packProjectResources(watch.ProjectResources)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("assigned_policy", packAssignedPolicies(watch.AssignedPolicies)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+func resourceXrayWatchUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := unpackWatch(d)
+	resp, err := m.(*resty.Client).R().SetBody(watch).Put("xray/api/v2/watches/" + d.Id())
+	if err != nil {
+		if resp != nil && resp.StatusCode() == http.StatusNotFound {
+			log.Printf("[WARN] Xray watch (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	d.SetId(watch.GeneralData.Name)
+	resourceXrayWatchRead(ctx, d, m)
```

Comment:
same problem as before - you never set `diags` and you don't return the results of the `read` simply `return resourceXrayWatchRead(ctx,d,m)`